### PR TITLE
Use system failure for node deletion or infra level issues

### PR DESCRIPTION
**Pull request by surindersinghp**

### CODE REVIEW

File: go/tasks/pluginmachinery/flytek8s/pod_helper.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func DemystifyPending(status v1.PodStatus) (pluginsCore.PhaseInfo, error) {
 								// So be default if the container is not waiting with the PodInitializing/ContainerCreating
 								// reasons, then we will assume a failure reason, and fail instantly
 								t := c.LastTransitionTime.Time
-								return pluginsCore.PhaseInfoRetryableFailure(c.Reason, c.Message, &pluginsCore.TaskInfo{
+								return pluginsCore.PhaseInfoSystemRetryableFailure(c.Reason, c.Message, &pluginsCore.TaskInfo{
```
Comment (akhurana001) [inline go/tasks/pluginmachinery/flytek8s/pod_helper.go]: default will always go to system failure

File: go/tasks/pluginmachinery/core/phase.go [hunk 2]
```diff
@@ -186,3 +189,7 @@ func PhaseInfoFailure(code, reason string, info *TaskInfo) PhaseInfo {
 func PhaseInfoRetryableFailure(code, reason string, info *TaskInfo) PhaseInfo {
 	return PhaseInfoFailed(PhaseRetryableFailure, &core.ExecutionError{Code: code, Message: reason}, info)
 }
+
+func PhaseInfoSystemRetryableFailure(code, reason string, info *TaskInfo) PhaseInfo {
+	return PhaseInfoFailed(PhaseRetryableFailure, &core.ExecutionError{Code: code, Message: reason, Kind: core.ExecutionError_SYSTEM}, info)
```
Comment (akhurana001) [inline go/tasks/pluginmachinery/core/phase.go]: PhaseSystemRetryableFailure not used

File: go/tasks/pluginmachinery/flytek8s/pod_helper.go (hunk 1 above)
Comment (surindersinghp) [reply go/tasks/pluginmachinery/flytek8s/pod_helper.go]: yep

Comment (EngHabu) [inline go/tasks/pluginmachinery/flytek8s/pod_helper.go]: Should we convert these into system errors:
https://github.com/lyft/flyteplugins/blob/779aab5d88b0add69358ab8269dee17fc87e1218/go/tasks/plugins/hive/execution_state.go#L108

https://github.com/lyft/flyteplugins/blob/c7dd2ebbc8982984422a91394d6490a76d2c2e7f/go/tasks/plugins/array/core/state.go#L215

File: go/tasks/pluginmachinery/core/phase.go [hunk 3]
```diff
@@ -186,3 +187,7 @@ func PhaseInfoFailure(code, reason string, info *TaskInfo) PhaseInfo {
 func PhaseInfoRetryableFailure(code, reason string, info *TaskInfo) PhaseInfo {
 	return PhaseInfoFailed(PhaseRetryableFailure, &core.ExecutionError{Code: code, Message: reason}, info)
```
[the remark this answers is not in the document]
Comment (surindersinghp) [reply go/tasks/pluginmachinery/core/phase.go]: This is the default: user + not-assigned (whatever we have today - system ones).

File: go/tasks/pluginmachinery/flytek8s/pod_helper.go (hunk 1 above)
Comment (surindersinghp) [reply go/tasks/pluginmachinery/flytek8s/pod_helper.go]: Done

File: go/tasks/plugins/array/core/state.go [hunk 4]
```diff
@@ -212,7 +212,7 @@ func MapArrayStateToPluginPhase(_ context.Context, state *State, logLinks []*idl
 		if state.GetExecutionErr() != nil {
 			phaseInfo = core.PhaseInfoFailed(core.PhasePermanentFailure, state.GetExecutionErr(), nowTaskInfo)
 		} else {
-			phaseInfo = core.PhaseInfoFailure(ErrorK8sArrayGeneric, state.GetReason(), nowTaskInfo)
+			phaseInfo = core.PhaseInfoSystemRetryableFailure(ErrorK8sArrayGeneric, state.GetReason(), nowTaskInfo)
```
Comment (surindersinghp) [inline go/tasks/plugins/array/core/state.go]: @EngHabu I updated this based on your feedback :) Do we really want to retry these failures? Not sure what is being marked as PhasePermanentFailure

Comment (EngHabu) [reply go/tasks/plugins/array/core/state.go]: I think you are right here, this should continue to be a regular failure.. sorry for the false alarm

**Pull request merged**